src/main/java/data/SheetsConverter.java

```diff
@@ -61,59 +62,69 @@ public void writeToSheet(ArrayList<Vendor> vendorList, Sheets sheetsService)
       .values().update(SPREADSHEET_ID, VENDOR_RANGE, vendorBody)
       .setValueInputOption(INPUT_OPTION).execute();
   }
-
+ 
   /** Clear all data from the Account and Vendor spreadsheets. */
   public void clearSheets(Sheets sheetsService) throws IOException {
     ClearValuesRequest requestBody = new ClearValuesRequest();
-
+ 
     ClearValuesResponse clearAccount =  sheetsService.spreadsheets().values()
       .clear(SPREADSHEET_ID, TAB_1, requestBody).execute();
     ClearValuesResponse clearVendor = sheetsService.spreadsheets().values()
       .clear(SPREADSHEET_ID, TAB_2, requestBody).execute();
   } 
-
+ 
   /** Store the Accounts(value) associated to its Vendor(key) in a HashMap. */
   public HashMap<String, ArrayList<Account>> getAllAccounts(
       ArrayList<Vendor> vendors) {      
-
+ 
     HashMap<String, ArrayList<Account>> allAccounts =  
       new HashMap<String, ArrayList<Account>>();
-
+ 
     for(int i = 0; i < vendors.size(); i++) {
       allAccounts.put(vendors.get(i).getVendorID(), vendors.get(i).getAccounts());
     }
-
+ 
     return allAccounts; 
   }
-
+ 
   public List<List<Object>> buildAccountSheetBody(
       HashMap<String, ArrayList<Account>> accounts) {
 
-    // TODO: @cfloeder Sort the accounts by some criteria x. 
     List<List<Object>> accountSheetData = new ArrayList<List<Object>>();
-    accountSheetData.add(new ArrayList<Object>(Account.getAccountSheetHeader()));
-
-    for(Map.Entry<String, ArrayList<Account>> entry : accounts.entrySet()) { 
-      ArrayList<Account> accountList = entry.getValue();
-      for(int i = 0; i < accountList.size(); i++) {
-        accountSheetData.add(new ArrayList<Object>(accountList.get(i)
-        .getAccountSheetsRow(entry.getKey())));
+ 
+    // Sort the Account Sheet aphabetically by VendorID.
+    ArrayList<String> sortedKeys = sortKeys(accounts);
+ 
+    for(int i = 0; i < sortedKeys.size(); i++) {
+      ArrayList<Account> accountList = accounts.get(sortedKeys.get(i));
+      for(int j = 0; j < accountList.size(); j++) {
+        accountSheetData.add(new ArrayList<Object>(accountList.get(j)
+          .getAccountSheetsRow(sortedKeys.get(i))));
```

Comment:
Not a Java expert, but did you consider using something like a LinkedHashMap or TreeMap to be able to order entries? Building an entirely new map doesn't seem too efficient here

Reply:
I don't build a new map here. I use the existing map that gets passed into the method